celery/bin/worker.py

```diff
@@ -161,7 +197,6 @@ def detach(path, argv, logfile=None, pidfile=None, uid=None,
               '--concurrency',
               type=int,
               metavar="<concurrency>",
-              callback=lambda ctx, _, value: value or ctx.obj.app.conf.worker_concurrency,
```

Comment:
This cannot be removed. The default value must be derived from the app if it is not supplied.